File: src/main/java/no/nav/samordning/innlasting/SamordningHendelseMapper.java
package no.nav.samordning.innlasting;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import no.nav.samordning.schema.SamordningHendelse;

class SamordningHendelseMapper {

    private static ObjectMapper objectMapper = new ObjectMapper();

    static String toJson(SamordningHendelse samordningHendelse) {
        try {
            return objectMapper.writeValueAsString(samordningHendelse.toString());
        } catch (JsonProcessingException e) {
            throw new RuntimeException("Mapping of hendelse failed: " + samordningHendelse.toString(), e);
        }
    }
}

File: src/main/kotlin/no/nav/samordning/innlasting/SamordningHendelseMapper.kt
package no.nav.samordning.innlasting

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import no.nav.samordning.schema.SamordningHendelse

internal object SamordningHendelseMapper {

    private val objectMapper = ObjectMapper()

    fun toJson(samordningHendelse: SamordningHendelse) = try {
        objectMapper.writeValueAsString(samordningHendelse.toString())
    } catch (e: JsonProcessingException) {
        throw RuntimeException("Mapping of hendelse failed: $samordningHendelse", e)
    }
}
